jupyterhub/scopes.py

```diff
@@ -629,3 +630,95 @@ def describe_raw_scopes(raw_scopes, username=None):
             }
         )
     return descriptions
+
+
+_custom_scope_pattern = re.compile(r"^custom:[a-z0-9]+[a-z0-9_\-\*:]+[a-z0-9_\*]$")
```

Comment:
> Is it relevant to check for empty strings between the colons, like custom:consideratio:read::books also?

I don't think so. After `custom:` the rest is up to the extension to interpret. It is not a sequence of fields, it is an opaque string. The only thing important to JupyterHub is that it can't get parsed incorrectly, e.g. including `!user=...`.